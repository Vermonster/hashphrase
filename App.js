import React from 'react';
import { createAppContainer, createStackNavigator } from 'react-navigation';
import { I18nextProvider } from 'react-i18next';
import { DefaultTheme, Provider as PaperProvider } from 'react-native-paper';
import i18n from './src/navigation/i18n/i18n';
import CreatePassword from './src/navigation/screens/CreatePassword';
import TipsPage from './src/navigation/screens/TipsPage';

const theme = {
  ...DefaultTheme,
  roundness: 2,
  colors: {
    ...DefaultTheme.colors,
    primary: '#E07926',
    accent: '#212A59',
    error: '#DB4437',
    text: '#444545',
    placeholder: '#949494',
    disabled: '#212A59',
  },
};

const AppStackNavigator = createStackNavigator(
  {
    PasswordScreen: {
      screen: CreatePassword,
    },
    TipsPage: { screen: TipsPage },
  },
  {
    initialRouteName: 'PasswordScreen',
    headerLayoutPreset: 'left',
  },
);

const AppContainer = createAppContainer(AppStackNavigator);

const App = () => (
  <I18nextProvider i18n={i18n}>
    <PaperProvider theme={theme}>
      <AppContainer />
    </PaperProvider>
  </I18nextProvider>
);

export default App;
