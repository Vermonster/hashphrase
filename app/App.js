import { createAppContainer, createStackNavigator } from 'react-navigation';
import CreatePasswordScreen from './src/navigation/screens/CreatePasswordScreen';
import InfoScreen from './src/navigation/screens/InfoScreen';

const RootStack = createStackNavigator({
  Home: CreatePasswordScreen,
  Info: InfoScreen,
},
{
  initialRouteName: 'Home',
});

const AppContainer = createAppContainer(RootStack);

export default AppContainer;
