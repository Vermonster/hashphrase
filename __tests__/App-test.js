/* @jest-environment jsdom */

import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';

jest.mock('react-navigation', () => ({
  createAppContainer: jest.fn().mockReturnValue(
    // eslint-disable-next-line
    function NavigationContainer(props) { return null; }
  ),
  createStackNavigator: jest.fn(),
}));

describe('<App />', () => {
  it('should render the same snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
