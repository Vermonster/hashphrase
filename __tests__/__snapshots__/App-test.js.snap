// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render the same snapshot 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
